### PR TITLE
Filling in some gaps in the introduction

### PR DESCRIPTION
**Pull request by tobyhodges**

### CODE REVIEW

File: _episodes/01-introduction.md [hunk 1]
```diff
@@ -80,7 +80,27 @@ are called _dynamic_.
 This lesson focuses on static sites and tools that can be used to create them,
 known as **Static Site Generators**.
 
-TODO: add discussion of when SSGs are useful and when not.
+![Pages can be created by static server-side generation, where HTML is generated once on the server and doesn't change thereafter, by dynamic server-side generation, where the server can update and send new HTML based on requests from the browser, and client-side generation, where HTML is generated by the browser itself by processing source data from the server(s) with previously-loaded Javascript code.](/fig/page-generation-js4ds.svg)
+_Figure 1.1: Page Generation Alternatives. This figure was originally published in [JavaScript for Data Science][js4ds], and is reproduced here with permission from the original author._
```
Comment (sstevens2) [inline _episodes/01-introduction.md:84]: Maybe just something with the local rendering but it isn't rendering the `js4ds` link right.  I can't see anything wrong with it though.

**Review by sstevens2 — APPROVED** [review]
This looks like a great addition to me.  I left one comment but it might be due to my local render because I don't see anything wrong.

File: _episodes/01-introduction.md [hunk 2]
```diff
@@ -25,9 +25,9 @@ to render and display the webpage on our screen.
 which is just one reason why
 web developers are not the target audience of this tutorial.)
 
-The information displayed by the web browser is a combination of
-**HTML** - a hierarchical format describing the structural elements of the page and their raw content -
-**CSS** - an ordered set of styling instructions telling the browser how to organise and format that content -
+The page displayed by the web browser is the result of combining
+**HTML** - a hierarchical format describing the structural elements of the page and their raw content - with
+**CSS** - an ordered set of styling instructions telling the browser how to the content should be organised and formatted -
```
Comment (unode) [inline _episodes/01-introduction.md:30]:
```suggestion
**CSS** - an ordered set of styling instructions telling the browser how the content should be organised and formatted -
```

Comment (unode) [inline _episodes/01-introduction.md]:
```suggestion
**CSS** - an ordered set of styling instructions telling the browser how the content should be organised and formatted -
```

File: _episodes/01-introduction.md [hunk 3]
```diff
@@ -80,7 +80,27 @@ are called _dynamic_.
 This lesson focuses on static sites and tools that can be used to create them,
 known as **Static Site Generators**.
 
-TODO: add discussion of when SSGs are useful and when not.
+![Pages can be created by static server-side generation, where HTML is generated once on the server and doesn't change thereafter, by dynamic server-side generation, where the server can update and send new HTML based on requests from the browser, and client-side generation, where HTML is generated by the browser itself by processing source data from the server(s) with previously-loaded Javascript code.](/fig/page-generation-js4ds.svg)
+_Figure 1.1: Page Generation Alternatives. This figure was originally published in [JavaScript for Data Science][js4ds], and is reproduced here with permission from the original author._
+
+Static-generated sites are a great choice when the information you want to
+display on a website is the same regardless of who visits your site
+and when they visit,
```
Comment (unode) [inline _episodes/01-introduction.md]:
```suggestion
display on a website is the same regardless of who visits your site,
when they visit,
```

File: _episodes/01-introduction.md [hunk 4]
```diff
@@ -80,7 +80,27 @@ are called _dynamic_.
...
+and when it's unlikely to need to change very often.
+This makes Static Site Generators a good choice for sites that provide
+documentation or lesson content like this page:
+the aim of the page is to deliver the same information,
+and the visitor arrives,
+(hopefully) finds and reads what they need,
+and leaves feeling happy and fulfilled.
```
Comment (unode) [inline _episodes/01-introduction.md]:
```suggestion
the aim of the page is to always deliver the same information.
The visitor arrives,
(hopefully) finds and reads what they need
and leaves feeling happy and fulfilled.
```

**Review by unode — CHANGES_REQUESTED** [review]
Minor typos, otherwise :+1:

**Pull request merged**